src/Compiler/Compiler.cpp

```diff
@@ -1461,33 +1461,248 @@ bool Compiler::RegisterCommands(TclInterpreter* interp, bool batchMode) {
+      cmd += "\n";
+    }
+
+    if (raiseGtkWindow) {
+      // Add a window raise command before the command to try to bring gtkwave
+      // to focus
+      cmd = "gtkwave::presentWindow\n" + cmd;
+    }
+
+    // Send command to GTKWave's stdin
+    process->write(cmd.c_str());
+    process->waitForBytesWritten();
+  }
+}
+
+// This will return a pointer to the current gtkwave process, if no process is
+// running, one will be started
+QProcess* Compiler::GetGTKWaveProcess() {
+  if (!m_gtkwave_process) {
+    m_gtkwave_process = new QProcess();
+
+    // Get a path to our local gtkwave binary
+    auto binPath = GlobalSession->Context()->BinaryPath();
+    auto exePath = binPath / "gtkwave" / "bin";
+    QString wd = QString::fromStdString(exePath.string());
+    // Need to be in the gtkwave bin dir to resolve some dependency issues
+    m_gtkwave_process->setWorkingDirectory(wd);
+
+    // Enable process control via tcl interface over std in with --wish
+    QString wishArg = "--wish";
+    QStringList args{wishArg};
+
+    // Start GTKWave Process
+    // invoking ./gtkwave to ensure we load the local copy
+    m_gtkwave_process->start("./gtkwave", args);
+    if (m_gtkwave_process->waitForStarted()) {
+      // Clear pointer on process close
+      QObject::connect(
+          m_gtkwave_process,
+          // static_cast required due to overload of signal params
+          static_cast<void (QProcess::*)(int, QProcess::ExitStatus)>(
+              &QProcess::finished),
+          [this](int exitCode, QProcess::ExitStatus exitStauts) {
+            if (m_gtkwave_process) {
+              m_gtkwave_process->deleteLater();
+              m_gtkwave_process = nullptr;
+            }
+          });
+
+      // Listen to stdout
+      QObject::connect(
+          m_gtkwave_process, &QProcess::readyReadStandardOutput, [this]() {
+            // If we want to listen for return values from gtkwave
+            // getters, this is where we should check the value.
+
+            // Possible way of returning getter results:
+            // `wave_cmd puts \[gtkwave::getWaveWidth\]`
+            // also see installGTKWaveHelpers() for fine control
+
+            // Read stdout data
+            QByteArray data = m_gtkwave_process->readAllStandardOutput();
+            QString trimmed = data.trimmed();
+            // std::cout << "stdout: " << trimmed.toStdString() << std::endl;
+
+            // Listen for the wish interface being opened
+            if (trimmed.startsWith("Interpreter id is")) {
+              // Install extra tcl helpers
+              installGTKWaveHelpers();
+            }
+          });
+
+      // Listen to stderr
+      QObject::connect(m_gtkwave_process, &QProcess::readyReadStandardError,
+                       []() {
+                         // TODO @skyler-rs nov2022 This should be logged once
+                         // we have a logging system
+
+                         // QByteArray data =
```

Comment:
Debug ?

Reply:
Yes, we don't have a way of logging errors just for devs so this gets uncommented often when debugging.